blockchain/state_manager/src/lib.rs

```diff
@@ -234,4 +243,35 @@ where
             )),
         }
     }
+
+    pub fn lookup_id<'a>(&'a self, addr: &Address, ts: &Tipset) -> Result<Address, Error> {
+        let state_tree = StateTree::new_from_root(self.bs.as_ref(), ts.parent_state())?;
+        state_tree.lookup_id(addr).map_err(Error::State)
+    }
+
+    pub fn market_balance<'a>(
+        &'a mut self,
+        addr: &Address,
+        ts: &Tipset,
+    ) -> Result<MarketBalance, Error> {
+        let market_state: market::State =
+            self.load_actor_state(&*STORAGE_MARKET_ACTOR_ADDR, ts.parent_state())?;
+
+        let new_addr = self.lookup_id(addr, ts)?;
+
+        let et = BalanceTable::from_root(self.bs.as_ref(), &market_state.escrow_table).unwrap();
+
+        let mut out = MarketBalance::default();
+
+        if et.has(&new_addr).is_ok() {
```

Comment:
this will return true if the result of the function is `Ok` and not if the boolean in the Ok is true. (will give a false positive when returns `Ok(false)`) so should probably just replace `is_ok` with `?`